modules/library/jdbc/src/main/java/org/geotools/data/jdbc/FilterToSQL.java

```diff
@@ -746,6 +770,21 @@ protected void visitBinaryComparisonOperator(BinaryComparisonOperator filter, Ob
 
         Expression left = filter.getExpression1();
         Expression right = filter.getExpression2();
+
+        // shortcut for IN case, to have a more natural encoding for the common/sane case of 
+        // "in(property, v1, v2, ... vn) = true" into "property in (v1, v2, ..., vn) 
+        if ("=".equals(extraData) || "!=".equals(extraData)) {
+            if (left instanceof Function && right instanceof Literal && InFunction.isInFunction((Function) left) && 
```

Comment:
Method ``isInFunction`` could accept an ``Expression`` directly and perform the ``instance of`` internally.

Reply:
Hmm.. yep